src/providers/noop_test.rs

```diff
@@ -33,9 +33,7 @@ mod noop_test {
     fn send_notification() {
         let mut provider = NoopProvider::new();
 
-        provider
-            .send_notification(PUSH_TOKEN.to_string(), EXAMPLE_MESSAGE.to_string())
-            .expect("Failed to send notification");
+        provider.send_notification(PUSH_TOKEN.to_string(), EXAMPLE_MESSAGE.to_string());
```

Comment:
Maybe this should now be an async test to be able to await `send_notification()` result?

I don't think that `send_notification()` would event execute without an `await`.